sympy/physics/continuum_mechanics/beam.py

```diff
@@ -180,7 +195,22 @@ def second_moment(self):
 
     @second_moment.setter
     def second_moment(self, i):
-        self._second_moment = sympify(i)
+        self._cross_section = None
+        if isinstance(i, GeometryEntity):
+            raise ValueError("To update cross-section geometry use `cross_section` attribute")
+        else:
+            self._second_moment = sympify(i)
+
+    @property
+    def cross_section(self):
+        """Cross-section of the beam"""
+        return self._cross_section
+
+    @cross_section.setter
+    def cross_section(self, s):
+        if s:
+            self._second_moment = s.second_moment_of_area()[0]
```

Comment:
Being that this selects the Ixx value, I think the documentation should reflect this. Say something like "If a Geometry object is used, it is assumed that the X axis of the object is aligned with the bending axis.".

Reply:
I have made the change.  Please have a look

Reply:
Looks good. It can be merged with the tests pass.